src/components/Calculator.js

```diff
@@ -1,67 +1,59 @@
-import React, { PureComponent } from 'react';
+import React, { useState } from 'react';
 import './Calculator.css';
 import calculate from '../logic/calculate';
 
-export default class Calculator extends PureComponent {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: null,
-      next: null,
-      operation: null,
-    };
-  }
+function Calculator() {
```

Comment:
[OPTIONAL] You have done an excellent job using hooks! I would suggest using arrow functions as this is a better practice in the industry.